docs/topics/sgx/sgx.md

```diff
@@ -0,0 +1,184 @@
+
+## Deploying the SGX device plugin
+
+You can install the SGX device plugin which surfaces the usage of Intel SGX’s Encrypted Page Cache (EPC) RAM as a schedulable resource for Kubernetes. This allows you to schedule pods that use the [Open Enclave SDK](https://github.com/openenclave/openenclave) or [Intel SGX SDK](https://github.com/intel/linux-sgx) onto hardware which supports Trusted Execution Environments.
+
+### Device plugin installation
+
+Using kubectl, deploy device plugin DaemonSet: 
+1. For kubernetes versions before v1.17, use: <br>
+    kubectl create -f [device-plugin-before-k8s-1-17.yaml](device-plugin-before-k8s-1-17.yaml)
+2. For kubernetes v1.17 and onwards, use: <br> 
+    kubectl create -f [device-plugin.yaml](device-plugin.yaml)
+
+Confirm that the DaemonSet pods are running on each Intel SGX enabled node as follows:
+
+```bash
+$ kubectl get pods -n kube-system -l app=sgx-device-plugin
+```
+
+```bash
+NAME                         READY   STATUS    RESTARTS   AGE
+sgx-device-plugin-7d5l8   1/1     Running   0          12m
+sgx-device-plugin-jzhk9   1/1     Running   0          12m
+```
+
+Confirm that the device plugin is advertising the available EPC RAM to the Kubernetes scheduler by running the following command:
+
+```bash
+$ kubectl get nodes <node-name> -o yaml
+```
+
+Under the status field you should see the total allocable resources with a name of `kubernetes.azure.com/sgx_epc_mem_in_MiB` 
+```bash
+<snip>
+status:
+  allocatable:
+    kubernetes.azure.com/sgx_epc_mem_in_MiB: "82"
+<snip>
+```
+
+## Scheduling Pods to TEE enabled Hardware
+
+The following pod specification demonstrates how you would schedule a pod to have access to a TEE by defining a limit on the specific EPC memory that is advertised to the Kubernetes scheduler by the device plugin
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: oe-deployment
+spec:
+  selector:
+    matchLabels:
+      app: oe-app
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: oe-app
+    spec:
+      tolerations:
+      - key: kubernetes.azure.com/sgx_epc_mem_in_MiB
+        operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: <image_name>
+        image: <image_reference>
+        command: <exec>
+        resources:
+          limits:
+            kubernetes.azure.com/sgx_epc_mem_in_MiB: 10
+```
+
+You can use the following test workload to confirm that your cluster is correctly configured ([Dockerfile](https://github.com/microsoft/openenclave-aks/blob/master/k8s-sgxtest/Dockerfile) for sgx-test):
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: sgx-test
+  labels:
+    app: sgx-test
+spec:
+  template:
+    metadata:
+      labels:
+        app: sgx-test
+    spec:
+      containers:
+      - name: sgxtest
+        image: oeciteam/sgx-test:1.0
+        command: ["/samples/helloworld/host/helloworldhost", "/samples/helloworld/enclave/helloworldenc.signed"]
+        volumeMounts:
```

Comment:
volume mount is no longer needed, as its using the device plugin

Reply:
in fact a bunch of this is not needed because of the daemonset. Especially as that will be there by default going forwards I think this can be cleaned up.